### ISSUE TITLE
Add DocBlock templates

### ISSUE THREAD

**Issue by mvriel** [phpDocumentor/phpDocumentor#42]
Add DocBlock templates as implemented in phpDocumentor: http://manual.phpdoc.org/HTMLSmartyConverter/HandS/phpDocumentor/tutorial_phpDocumentor.howto.pkg.html#basics.docblocktemplate

**Comment by filips123** [comment]
Is there any update? Will DocBlock templates be supported in phpDocumentor 3?

**Comment by nelson6e65** [comment]
I don't think is a good idea. Maybe will be fine for generating HTML documentation, but will not be IDEs-like friendly.

**Comment by johnpancoast** [comment]
Hi @nelson6e65, can you elaborate on why you don't think this would be IDE friendly?

**Comment by johnpancoast** [comment]
I agree with both @thomshouse and @localheinz in that members and constants often benefit from this feature.

**Comment by mvriel** [comment]
Moved this onto our 'Wishlist' project board as a note so that it would not show up as an issue in our tracker. We are doing this for feature requests we cannot complete in a short timespan and which does not have a high priority to make sure the issue listing reflects work that we want to pick up now.

As for the reason why DocBlock templates have not returned after to so long:

This is not a trivial issue since the Parser we built does not track state and reflects the DocBlocks associated with elements and not free-floating DocBlocks such as these; to fix this; we'd need to do open-heart surgery on these parts of phpDocumentor and we do not feel that the time and effort needed for this weigh up.

I apologize if this inconveniences some, we, unfortunately, have to make choices with the small team that we have.
